Dockerfile

```diff
@@ -0,0 +1,18 @@
+# --- builder ---
+FROM golang:1.20.6-alpine3.17 as builder
+LABEL stage=builder
+RUN apk add git
+WORKDIR /build
+
+COPY go.* ./
+RUN go mod download
+
+COPY . ./
+RUN go build -ldflags "-X 'main.serverBuildString=pretendo.pikmin3.docker'" -v -o server
```

Comment:
Is there any way to make this build string a bit more dynamic? I'm unsure what limitations or options we have here

Reply:
Added a variable to the Dockerfile with a default value. This default value can be overridden via the --build-arg flag. Notes in README.

Reply:
That looks good. Would you be opposed to having an optional `docker` rule to the Makefile that runs these commands so it can pass the git-based build string? This wouldn't be part of the Docker instructions, it would be in the `make` instructions and be entirely optional. This doesn't have to be added in this PR I can do it later if that's fine by you

Reply:
Sounds fine to me. I'm not as familiar with `make` as you are, so it would probably be better if you added that.